src/Domain/CoordinateMaps/TimeDependent/Skew.cpp

```diff
@@ -0,0 +1,423 @@
+  auto& tan_sum = get<0, 2>(result);
+  tan_sum = tan_func[0] * source_coords[1] + tan_func[1] * source_coords[2];
+
+  get<0, 0>(result) += width_deriv[0] * tan_sum;
+  get<0, 1>(result) = width_deriv[1] * tan_sum + width * tan_func[0];
+  get<0, 2>(result) = width_deriv[2] * tan_sum + width * tan_func[1];
+
+  return result;
+}
+
+template <typename T>
+tnsr::Ij<tt::remove_cvref_wrap_t<T>, 3, Frame::NoFrame> Skew::inv_jacobian(
+    const std::array<T, 3>& source_coords, const double time,
+    const std::unordered_map<
+        std::string, std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>>&
+        functions_of_time) const {
+  return determinant_and_inverse(
+             jacobian(source_coords, time, functions_of_time))
+      .second;
+}
+
+template <typename T>
+tt::remove_cvref_wrap_t<T> Skew::get_delta(
+    const std::array<T, 3>& centered_source_coords) const {
+  return square(dot(centered_source_coords, center_)) +
+         dot(centered_source_coords, centered_source_coords) *
+             outer_radius_squared_minus_center_radius_squared_;
+}
+
+template <typename T>
+tt::remove_cvref_wrap_t<T> Skew::get_width(
+    const std::array<T, 3>& source_coords) const {
+  using ResultT = tt::remove_cvref_wrap_t<T>;
+  std::array<ResultT, 3> centered_source_coords{};
+  for (size_t i = 0; i < 3; i++) {
+    gsl::at(centered_source_coords, i) =
+        gsl::at(source_coords, i) - gsl::at(center_, i);
+  }
+  // Will be reused for result
+  ResultT lambda = (dot(centered_source_coords, center_) +
+                    sqrt(get_delta(centered_source_coords))) /
+                   outer_radius_squared_minus_center_radius_squared_;
+
+  ResultT& result = lambda;
+
+  for (size_t i = 0; i < get_size(result); i++) {
+    if (get_element(lambda, i) <= 0.0 and
+        equal_within_roundoff(get_element(lambda, i), 0.0)) {
+      get_element(result, i) = 1.0;
+    } else if (get_element(lambda, i) >= 1.0 and
+               equal_within_roundoff(get_element(lambda, i), 1.0)) {
+      get_element(result, i) = 0.0;
+    } else if (get_element(lambda, i) > 0.0 and get_element(lambda, i) < 1.0) {
+      get_element(result, i) = 0.5 * (1.0 + cos(M_PI * get_element(lambda, i)));
+    } else {
+      using ::operator<<;
+      const std::vector<double> bad_point{
+          get_element(centered_source_coords[0], i),
+          get_element(centered_source_coords[1], i),
+          get_element(centered_source_coords[2], i)};
+      ERROR("Skew map: Centered source coordinate "
+            << bad_point << " not in valid region. Lambda is "
+            << get_element(lambda, i) << ". Center is " << center_
+            << ". Outer radius is " << outer_radius_);
+    }
+  }
+
+  return result;
+}
+
+template <typename T>
+std::array<tt::remove_cvref_wrap_t<T>, 3> Skew::get_width_deriv(
+    const std::array<T, 3>& source_coords_cvref) const {
+  using ResultT = tt::remove_cvref_wrap_t<T>;
+  std::array<ResultT, 3> source_coords{};
```

Comment:
Why do you need this non-const variable? Seems dangerous. Please add a comment if this is really necessary.